### PR TITLE
Directed Graph functionality with tests, incomplete

### PR DESCRIPTION
**Pull request by AbrahamFergie**

### CODE REVIEW

File: spec/directedGraph_test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  {vertex1:'f', vertex2:'n', weight:3},{vertex1:'g', vertex2:'m', weight:1},
+  {vertex1:'g', vertex2:'o', weight:1},{vertex1:'h', vertex2:'m', weight:2},
+  {vertex1:'h', vertex2:'p', weight:2},{vertex1:'h', vertex2:'q', weight:1},
+  {vertex1:'i', vertex2:'r', weight:2},{vertex1:'j', vertex2:'k', weight:3}]
+
+  it('exists', () => {
+    expect(DirectedGraph).to.be.a('function')
+  })
+
+  context('addVertex() && hasVertex()', () => {
```
Comment (Anasauce) [inline spec/directedGraph_test.js:25]: pulled down your branch and this test isn't passing...can't quite figure out why. Schedule a coaching session and if we can't figure it out we can grab an SEP

File: src/directedGraph.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  //checks the directed graph if it has a particular direction
+  hasDirection(vertex1, vertex2, dir){
+    let currentVertex = this.edges[vertex1]
+    if(currentVertex === undefined){ return null }
+    let numOfDirs = currentVertex.length - 1
+    if(dir === undefined){
+      let dir = 1
+      if(currentVertex[dir].vertex2 === vertex2){
+        return true
+      }else{
```
Comment (Anasauce) [inline src/directedGraph.js:51]: no need for an ```else``` here since you have a ```return``` in your if conditionals

File: src/directedGraph.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    }
+    return false
+  }
+  //adds a vector connecting given vertices with a weight attached
+  addDirection(vertex1, vertex2, weight){
+    // console.log(this.edges[vertex1])
+    this.edges[vertex1].push({vertex1:vertex1, vertex2:vertex2, weight:weight})
+  }
+  //checks the directed graph if it has a particular direction
+  hasDirection(vertex1, vertex2, dir){
```
Comment (Anasauce) [inline src/directedGraph.js:43]: try refactoring this function, there are a lot of nested ```if/else```. See if it is parts of this function can be moved out into their own functions. There is the Single Responsibility Principle https://en.wikipedia.org/wiki/Single_responsibility_principle

File: src/directedGraph.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  removeDirection(vertex1, vertex2){}
+  //returns an array of all all free floating vertices
+  getSeparatedVertices(){}
+  //removes a particular vertex and all its directions
+  removeVertex(vertex){}
+  //returns the number of edges
+  count(){ return this.vertices }
+  //sketches an instance of the directed graph in its current state
+  printGraph(v) {
+    for ( var i = 0; i < this.vertices; ++i) {
```
Comment (Anasauce) [inline src/directedGraph.js:92]: var? ☹️

File: src/directedGraph.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      console.log(chalk.black.bgYellow(this.edges[v[i]][0]) + '->\n')
+      for (  var j = 0; j < this.edges[v[i]].length; j++ ) {
```
Comment (Anasauce) [inline src/directedGraph.js:94]: var? 😦

Comment (AbrahamFergie) [reply src/directedGraph.js:94]: haha i did forget about those. thank you

File: src/directedGraph.js (hunk 2 above)
Comment (AbrahamFergie) [reply src/directedGraph.js:51]: you right, nice one, thank you

**Pull request merged**